curta/src/chip/builder/arithmetic.rs

```diff
@@ -129,6 +129,7 @@ impl<L: AirParameters> AirBuilder<L> {
         AirInstruction::Assign(instr)
     }
 
+    // could u add more comment here? here it set expression to equal to data value, assign expression to data, the direction matter.
     #[inline]
     pub fn set_to_expression_transition<T: Register>(
```

Comment:
this function simultaenous constrains data to expression while also writing it in the trace.

note that using assert constrains but does not write the value to the trace.

Reply:
how does it write value to trace, thought this constraint is independent of trace generation step (unlike circom's <== / >==)? do u mean it constrain certain value for certain register? actually i am indeed confused on how this differ from assert, it seem assert work on abstract arithmetic expression while set work with register and arithmetic expression..